internal/storage/db_storage.go

```diff
@@ -0,0 +1,100 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"time"
+
+	_ "github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgconn"
+)
+
+type DBStorage struct {
+	db *sql.DB
+}
+
+func (ds *DBStorage) init(ctx context.Context, databaseDSN string) (err error) {
+	ds.db, err = sql.Open("pgx", databaseDSN)
+	if err != nil {
+		return
+	}
+	err = ds.migrate(ctx)
+	return
+}
+
+func (ds *DBStorage) migrate(ctx context.Context) (err error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Second)
+	defer cancel()
+
+	_, err = ds.db.ExecContext(ctx, `
+		CREATE TABLE IF NOT EXISTS urls (
+		    id uuid PRIMARY KEY DEFAULT gen_random_uuid(),
+		    short_url VARCHAR NOT NULL UNIQUE,
+		    url VARCHAR NOT NULL UNIQUE
+		    )
+		`)
+	return
+}
+
+func (ds *DBStorage) Store(ctx context.Context, Data string) (ID string, duplicate bool) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Second)
```

Comment:
CLARIFICATION: why do you need to think about it here?
You already have `context`.